File: flashscore/flashscore.com.ua/flashscore.com.ua-hockey/start1.php
<?php

ini_set('date.timezone', 'Europe/Moscow');
setlocale(LC_ALL, 'ru_RU.CP1251', 'rus_RUS.CP1251', 'Russian_Russia.1251');
error_reporting(E_ALL | E_STRICT) ;
ini_set('display_errors', 'On');		
set_time_limit(0); //устанавливаем время на безлимит
flush ();

$count = 0;
$count_day = 3;
$day_lockfile_delete = 2;

//$date = date("Y-m-d H:i:s");
//$date = getdate (time());

$time = time();
$lockfile = getcwd () .'/lockfile.txt';

#очистка от старого лок файла
if (is_file ($lockfile)) {
	$fh = fopen ($lockfile, 'r') or die ();
	while ($str = fgets ($fh)) {
		$temp = preg_split ('/\t/', $str);
		
		if (count ($temp) > 0) {
			foreach ($temp as $key => $value) {
				$value = str_replace (array ("\n", "\t", "\r"), '', $value);
				$value = trim ($value);
			}
			$time_lockfile = $temp[0];
		}
	}
	fclose ($fh);	

	$day = ($time - $time_lockfile) / 3600/24;
	$day = (int)$day;
	
	if ($day > $day_lockfile_delete) {
		unlink ($lockfile) or die;
	}
}



if (!is_file ($lockfile)) {

	$fh = fopen ($lockfile, 'w') or die ();
	fwrite ($fh, time () ."\t". 1);
	fclose ($fh);
	
	// $url_array = array ();
	// $time = time ();
	// while ($count_day > 0) {
	
		// $date = getdate ($time);
		// //print_r ($date);
		
		// $time = $time - 24*3600; //смотрим назад
		// //$time = $time + 24*3600; //смотрим вперед
		
		// $count_day--;
		
		// $date_0 = array ();
		// array_push ($date_0, $date['year']);
		// array_push ($date_0, $date['mon']);
		// array_push ($date_0, $date['mday']);
		
		// foreach ($date_0 as $key => $value) {
			// if (strlen ($value) < 2) {
				// $date_0 [$key] = '0'.$value;
			// }
		// }
		
		// //$url = 'http://scoresway.com/matches/'.$date_0[0].'/'.$date_0[1].'/'.$date_0[2].'/';
		// $url = 'http://www.scoresway.com/?sport=baseball&page=matches&date='.$date_0[0].'-'.$date_0[1].'-'.$date_0[2];
		// array_push ($url_array, $url);
	// }
	
	// array_push ($url_array, 1);
	// if (count ($url_array) > 0) {
		// $file = getcwd () .'/city/001.xls';
		
		// $fh = fopen ($file, 'w') or die ();
		// foreach ($url_array as $value) {
			// fwrite ($fh, $value ."\t". 1 ."\n");
		// }
		// fclose ($fh);
	// }
	
	
	$workdir0 = getcwd ();
	$workdir1 = getcwd () .'/city';
	$workdir2 = getcwd () .'/graber1';
	$workdir3 = getcwd () .'/graber2';
	

	$file_array = array ();
	$dh = opendir ($workdir1) or die;
	while ($file = readdir ($dh)) {
		if ($file != '.' and $file != '..') {
			$file = $workdir1.'/'.$file;
			$pattern = '/xls$/';
			
			if (preg_match ($pattern, $file)) {
				print ++$count ."\n";
				
				$fh = fopen ($file, 'r') or die ();
				while ($str = fgets ($fh)) {
					$str = str_replace(array("\r","\n"), '', $str);
					
					array_push ($file_array, $str);
				}
				fclose ($fh);
			}
		}
	}
	closedir ($dh);
	
	if (count ($file_array) > 0) {
		foreach ($file_array as $file_array_value) {
			$t = preg_split ('/\t/ui',$file_array_value);
			
			//graber1
			// if (is_dir ($workdir2)) {
				// chdir ($workdir2);
		
				// // $date = getdate (time());		
				
				// // $date1 = $date['year'];
				// // $date2 = $date1--;
				
				// $file = getcwd () .'/xls/001.xls';
				// $fh = fopen ($file, 'w') or die ();
				// fwrite ($fh, $t[0] ."\t". 1 . "\n");
				// fclose ($fh);
				
				// $file_start = 'start1.cmd';
				// if (is_file ($file_start) and is_dir ('c:\windows')) {
					// system  ($file_start);
				// }
				
				// $file_start = './start1';
				// if (is_file ($file_start) and !is_dir ('c:\windows')) {
					// system  ($file_start);
				// }
				
				// chdir ($workdir0);
			// }
			
			
			
			//graber2
			if (is_dir ($workdir3)) {
				chdir ($workdir3);
		
				$file = getcwd () .'/xls/001.xls';
				$fh = fopen ($file, 'w') or die ();
				fwrite ($fh, $t[0] ."\t". 1 . "\n");
				fclose ($fh);
				
				$file_start = 'start1.cmd';
				if (is_file ($file_start) and is_dir ('c:\windows')) {
					system  ($file_start);
				}
				
				$file_start = './start1';
				if (is_file ($file_start) and !is_dir ('c:\windows')) {
					system  ($file_start);
				}
				
				chdir ($workdir0);
			}
			
		}
	}

	unlink ($lockfile) or die ();
	
}

?>

File: soccerway.com/soccerway.com/start1.pl
#!/usr/bin/env perl
use strict; 
use warnings;
use Cwd;
use lib getcwd().'/graber';
use clear_str;
use Date::Calc qw (Date_to_Time Time_to_Date);


my $workdir0 = getcwd (); 
my $workdir1 = getcwd (). '/graber'; 
my $workdir2 = getcwd (). '/city'; 


my $day_lock_file_delete = 2;
my $count =  3;
my $time = time ();
my $time_lockfile = time ();
my $lockfile_graber = 'lock_graber.txt'; 

#проверка лок файла на старость. принудительное удаление.
if (-f $lockfile_graber) {
	open (FILE, $lockfile_graber) or die;
	while (<FILE>) {
		if ($_ =~ /\d/) {
			my @temp = split ("\t", $_);
			if (scalar (@temp) == 2) {
				foreach (@temp) {
					$_ =~ s/^\s+//;
					$_ =~ s/\s+$//;
				}
				$time_lockfile = $temp[0];
			}
		}
	}
	close (FILE); 
	
	my $day = ($time - $time_lockfile)/3600/24;
	$day = int ($day);
	
	if ($day > $day_lock_file_delete) {
		unlink ($lockfile_graber);
	}
}



if (!-f $lockfile_graber) {
	open (FILE, '>'.$lockfile_graber) or die;
	print FILE time () ."\t".1 ."\n";
	close (FILE); 


	my @url = ();
	do {
		# http://int.soccerway.com/matches/2016/08/16/
		
		my ($year,$month,$day, $hour,$min,$sec) = Time_to_Date ($time);
		my @date = Time_to_Date($time);
		foreach (@date) {
			while (length ($_) < 2) {
				$_ = '0'.$_;
			}
		}
		
		my $url = 'http://int.soccerway.com/matches/'.$date[0].'/'.$date[1].'/'.$date[2].'/';
		push (@url, $url);
		
		$time = $time - 24*3600;
		#$time = $time + 24*3600;

		$count --;

	} while ($count > 0);


	if (scalar (@url) > 0) {
		
		my $file = getcwd () .'/city/001.xls';
		open (FILE, '>'.$file) or die (print $!);
		foreach (@url) {
			print FILE $_ ."\t". 1 ."\n";
		}
		close (FILE);

		my @file1 = ();
		opendir (DIR, $workdir2) or die (print $!);
		while (my $file1 = readdir (DIR)) {
			# print ++$count ."\n";
			
			if ($file1 =~ /xls$/) {
				$file1 = $workdir2 .'/'.$file1;
				
				open (FILE, $file1) or die (print $!);
				while (<FILE>) {
					my $array = [];
					@$array = split ("\t", $_);
					foreach (@$array) {
						$_ =~ s/\n+//g;
						$_ =~ s/\r+//g;
					}
					if (scalar (@$array) == 2 and $array -> [0] =~ /^http/) {
						push (@file1, $array);
					}
				}
				close (FILE);
			}
		}
		closedir (DIR); 

		$count = 0;
		if (scalar (@file1) > 0) {
				
			foreach (@file1) {
				my $time = time ();
				
				my $log_file_stdout = getcwd ().'/log/'. get_date ($time) .'_stdout.xls';
				my $log_file_stderr = getcwd ().'/log/'. get_date ($time) .'_stderr.xls';
				
				chdir ($workdir1); 
				my $file_url = $workdir1  .'/xls/001.xls';
					
				open (FILE, '>'.$file_url) or die (print $!);
				print FILE $_-> [0] ."\t". $_ ->[1] ."\n";
				close (FILE);
				
				my $file2 = 'start1.cmd'; 
				if (-f $workdir1 .'/'. $file2 and -d 'c:/windows') {
					#system $file2;
					# system $file2 . ' 1>'. $log_file_stdout .' 2>'.$log_file_stderr;
				}
				
				$file2 = './start1.pl' ; 
				if (-f $workdir1 .'/'. $file2 and !-d 'c:/windows') {
					system $file2;
					# system $file2 . ' 1>'. $log_file_stdout .' 2>'.$log_file_stderr;
				}
				
				chdir ($workdir0);
			}
		}
	}
	
	unlink ($lockfile_graber) or die (print $!);
	
}




sub get_date { 
	use  Date::Calc qw (Time_to_Date Date_to_Time);
	# my ($year,$month,$day, $hour,$min,$sec) = Time_to_Date(time());
	
	my $time = shift;
	my @date = Time_to_Date($time);
	foreach (@date) {
		while (length ($_) < 2) {
			$_ = '0'.$_;
		}
	}
	
	my $str = $date[0] .'-'.$date[1].'-'.$date[2] .'_'.$date[3].'-'.$date[4].'-'.$date[5];
	return $str;
}
